### PR TITLE
Add mcuboot to list of project dependencies

### PR DESCRIPTION
**Pull request by utzig**
@sjanc @michal-narajowski This enables https://github.com/apache/mynewt-core/pull/1763/ to pass...

### CODE REVIEW

**Comment by michal-narajowski** [comment]
@utzig Should we add this change to nimble project.yml too?

File: mynewt-core-project.yml [hunk 1]
```diff
@@ -21,9 +21,16 @@ project.name: "apache-mynewt-core"
...
 
 repository.apache-mynewt-nimble:
     type: github
     vers: 0.0.0
     user: apache
     repo: mynewt-nimble
+
+repository.mcuboot:
+    type: github
+    vers: 0.0.0
```
Comment (sjanc) [inline mynewt-core-project.yml]: I'm not sure if we should use mcuboot master for this,  also nimble yml needs similar change I guess

**Comment by utzig** [comment]
@michal-narajowski Not sure it's required, is it? If something breaks of course...

File: mynewt-core-project.yml (hunk 1 above)
Comment (utzig) [reply mynewt-core-project.yml]: Why not sure?

**Comment by utzig** [comment]
@michal-narajowski @sjanc I added to nimble projects as well, no idea if it's required but doesn't hurt anyway...

File: mynewt-core-project.yml (hunk 1 above)
Comment (sjanc) [reply mynewt-core-project.yml]: master is floating target and core release would rather depend on specific version than master branch of external project

I think we should use 1-latest for our tests too.

Comment (utzig) [reply mynewt-core-project.yml]: To be clear, the rationale of using latest masters on travis-ci also enables to catch possible build failures although it should not. We do not merge broken commits, so this should always build the test programs. No idea why setting on latest release would be any better. Latest release should be used by people's projects, travis-ci should use master.

**Comment by utzig** [comment]
I triggered a nimble master build, we'll see soon enough if it works!

**Pull request merged**